### PR TITLE
Uutils ucp cleanup and renaming of old Cp impl

### PR DESCRIPTION
**Pull request by dmatos2012**
# Description
Branches off #10097, which replaces the current `cp` command on nushell(and replaces is with `cp-old`), to the `cp` implementation using `uutils` instead. For a more detailed discussion and description, please refer to the original PR. This is mainly as the cleanup once the first one is agreed upon and merged if given the ok by the team.

# User-Facing Changes

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used` to check that you're using the standard code style
- [X] `cargo test --workspace` to check that all tests pass (on Windows make sure to [enable developer mode](https://learn.microsoft.com/en-us/windows/apps/get-started/developer-mode-features-and-debugging))
- [X] `cargo run -- -c "use std testing; testing run-tests --path crates/nu-std"` to run the tests for the standard library

# After Submitting

### CODE REVIEW

File: crates/nu-command/Cargo.toml [hunk 1]
```diff
@@ -88,6 +88,9 @@ toml = "0.7"
 unicode-segmentation = "1.10"
 ureq = { version = "2.7", default-features = false, features = ["charset", "gzip", "json", "native-tls"] }
 url = "2.2"
+# uu_cp = { path = "../../../coreutils/src/uu/cp"}
+# 0.0.21 not on crates.io yet?
+uu_cp = { git = "https://github.com/uutils/coreutils.git", rev = "862a2df"}
```
Comment (fdncred) [inline crates/nu-command/Cargo.toml:93]: Does this work for the release that was just published to crates.io?

Comment (dmatos2012) [reply crates/nu-command/Cargo.toml:93]: I tried, but apparently it looks like is still not published to `crates.io`. https://crates.io/crates/coreutils/versions

Comment (tertsdiepraam) [reply crates/nu-command/Cargo.toml:93]: Ah I see you noticed this too. We got rate-limited while publishing :) I'll try to publish the rest of the crates tomorrow.

**Comment by fdncred** [comment]
Closing this one since I think these changes were built into https://github.com/nushell/nushell/pull/10097.  Thanks again!

**Pull request closed without merging**